File: Project/Generic_DS_Lib/Stack.java
import java.lang.*;
import java.util.*;

class node
{
    public int data;
    public node next;       // struct node * next;
}

class StackDS
{
    public node Head;
    public int Count;

    public StackDS()
    {
        Head = null;
        Count = 0;
    }

    public void PUSH(int no)
    {
        node newn = new node();

        newn.data = no;
        newn.next = null;

        if(Head == null)
        {
            Head = newn;
        }
        else
        {
            newn.next = Head;
            Head = newn;
        }
        Count++;
    }

    public void POP()
    {
        int no;
        if(Count == 0)
        {
            System.out.println("Stack is empty");
            return;
        }
        else if(Count == 1)
        {
            no = Head.data;
            Head = null;
        }
        else
        {
            no = Head.data;
            node temp = Head;
            Head = Head.next;
            temp = null;
        }
        System.out.println("Removed element is : "+no);
        Count--;
    }

    public void Display()
    {
        node temp = Head;

        System.out.print("Elements of stack are : ");
        while(temp != null)
        {
            System.out.print("|"+temp.data+"|-");
            temp = temp.next;
        }
        System.out.println("NULL");
    }

    public int CountNode()
    {
        return Count;
    }   
}

class Stack
{
    public static void main(String arg[])
    {
        StackDS obj = new StackDS();

        obj.PUSH(11);
        obj.PUSH(21);
        obj.PUSH(51);

        obj.Display();
        System.out.println("Number of elements in stack are : "+obj.CountNode());

        obj.POP();
        obj.POP();

        obj.Display();
        System.out.println("Number of elements in stack are : "+obj.CountNode());

        obj.POP();
        obj.POP();

        obj.Display();
        System.out.println("Number of elements in stack are : "+obj.CountNode());
    }
}

File: Project/Generic_DS_Lib/Stack.c
// Stack

#include <stdio.h>
#include <stdlib.h>

/*
1.PUSH()
2.POP()
3.Count()
4.Display()
*/

struct node
{
    int data;
    struct node * next;
};

typedef struct node NODE;
typedef struct node * PNODE;
typedef struct node ** PPNODE;

void Display(PNODE Head)
{
    while(Head != NULL)
    {
        printf("|%d|->",Head->data);
        Head = Head->next;
    }
    printf("NULL\n");
}

int Count(PNODE Head)
{
    int iCnt = 0;

    while(Head != NULL)
    {
        iCnt++;
        Head = Head->next;
    }
    return iCnt;
}

void PUSH(PPNODE Head, int No)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));
    newn->data = No;
    newn->next = NULL;

    if(*Head == NULL)       
    {
        *Head = newn;
    }
    else                    
    {
        newn->next = *Head;
        *Head = newn;
    }
    printf("Entered element is : %d\n",No);
}

void POP(PPNODE Head)
{
    int no = 0;
    PNODE temp = NULL;

    if(*Head == NULL)
    {
        printf("Stack is empty\n");
        return;
    }
    else if((*Head)->next == NULL)
    {
        no = (*Head)->data;
        free(*Head);
        *Head = NULL;
        printf("Removed element is : %d\n",no);
    }
    else
    {
        temp = *Head;
        no = (*Head)->data;
        *Head = (*Head)->next;
        free(temp);
        printf("Removed element is : %d\n",no);
    }
}

int main()
{
    int Ret = 0;
    PNODE First = NULL;

    PUSH(&First,11);
    PUSH(&First,21);
    PUSH(&First,51);
    
    Display(First);
    Ret = Count(First);
    printf("Number of elements in stack are : %d\n",Ret);

    POP(&First);
    POP(&First);
    Display(First);
    Ret = Count(First);
    printf("Number of elements in stack are : %d\n",Ret);

    POP(&First);
    POP(&First);
    Display(First);
    Ret = Count(First);
    printf("Number of elements in stack are : %d\n",Ret);

    return 0;
}